src/backend/PortabilityService.AnalysisService/DependencyInjection/ApiPortDataModule.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Autofac;
+using Microsoft.Fx.Portability;
+using Microsoft.Fx.Portability.ObjectModel;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace PortabilityService.AnalysisService
+{
+    public class ApiPortDataModule : Module
+    {
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder.Register(LoadApiPortData).SingleInstance();
+            builder.RegisterType<GitCatalogRecommendations>().As<IApiRecommendations>().SingleInstance();
+        }
+
+        private ApiPortData LoadApiPortData(IComponentContext context)
+        {
+            var settings = context.Resolve<IServiceSettings>();
+
+            var breakingChanges = Directory.EnumerateFiles(settings.BreakingChangesPath, "*.md")
```

Comment:
Do we need to handle empty `settings.BreakingChangesPath` here, or let it bubbles up?

Reply:
I like letting the exception bubble up unhandled because I don't think there's a reasonable way to handle it and carry on; it signals a botched deployment.

Reply:
Also, note the readiness check won't return ok when no breaking changes are loaded, so the platform shouldn't route traffic to the service if the path is incorrect.

Reply:
Sounds good. A log might help but in this case it probably would be obvious what goes wrong.